### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by meta-jugal-kukreja**

### CODE REVIEW

File: SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
+package scf.session3.assignment;
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java]: Sir, we were told not to make main class in these assignments.

File: SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperationsClumpTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package scf.session3.assignment;
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperationsClumpTest.java]: Done

File: SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	/**
+	 * This method is used to calculate the length of the the largest mirror
+	 * section in an array
+	 * 
+	 * @param inputArray
+	 *            is the array containing all the inputs
+	 * @return {int} length of the largest mirror section
+	 */
+	public int lengthOfLargestMirrorSection(int inputArray[]) {
+		if (inputArray.length == 0) //Throwing Assertion Error if array size is 0
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java]: Sir,
Assertion Error handling with assert keyword : Done.
And in test class there is only function calling. There is no validation. That's why it is checked in the function

File: SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		return lengthOfMirror;
+	}
+
+	/**
+	 * This method calculates the number of clumps in the given array
+	 * @param inputArray is the array containing inputs
+	 * @return {int} number of clumps in the array
+	 */
+	public int calculateNumberOfClumps(int inputArray[]) {
+		if (inputArray.length == 0) //Throwing Assertion Error if array size is 0
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java]: Sir,
Assertion Error handling with assert keyword : Done.
And in test class there is only function calling. There is no validation. That's why it is checked in the function

File: SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	/**
+	 * This method is used to rearrange the array such that after every X there is Y. Condition is that X's position should be changed
+	 * @param inputArray is the array containing inputs
+	 * @param X is the value of variable X
+	 * @param Y is the value of variable Y
+	 * @return {int[]} rearranged array with the given conditions
+	 */
+	public int[] fixXYproblem(int inputArray[], int X, int Y) {
+		if (inputArray.length == 0) //Throwing Assertion Error if array size is 0
+			throw new AssertionError("Array size can't be zero");
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java]: Done

File: SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (meta-deepak-goyal) [reply SCF Session 4 Assignment/src/scf/session3/assignment/ArrOperations.java]: Ok

**Pull request merged**